source/UberRidesTests/TokenManagerTests.swift

```diff
@@ -26,210 +26,236 @@ import XCTest
-        
+
         XCTAssertFalse(TokenManager.deleteToken(identifier))
-        
+
         NSNotificationCenter.defaultCenter().removeObserver(self)
-        
+
         XCTAssertFalse(notificationFired)
     }
-    
+
     func testCookiesCleared_whenTokenDeleted() {
         guard let usUrl = NSURL(string: "https://login.uber.com"), let chinaURL = NSURL(string: "https://login.uber.com.cn")  else {
             XCTAssertFalse(false)
             return
         }
-        
+
         let cookieStorage = NSHTTPCookieStorage.sharedHTTPCookieStorage()
-        
+
         if let cookies = cookieStorage.cookies {
             for cookie in cookies {
                 cookieStorage.deleteCookie(cookie)
             }
         }
-        
-        
+
+
         cookieStorage.setCookies(createTestUSCookies(), forURL: usUrl, mainDocumentURL: nil)
         cookieStorage.setCookies(createTestChinaCookies(), forURL: chinaURL, mainDocumentURL: nil)
         NSUserDefaults.standardUserDefaults().synchronize()
         XCTAssertEqual(cookieStorage.cookies?.count, 4)
         XCTAssertEqual(cookieStorage.cookiesForURL(usUrl)?.count, 2)
         XCTAssertEqual(cookieStorage.cookiesForURL(chinaURL)?.count, 2)
-        
+
         let identifier = "testIdentifier"
         let accessGroup = "testAccessGroup"
-        
+
         let token = getTestToken()
-        
+
         keychain?.setAccessGroup(accessGroup)
         keychain?.setObject(token, key: identifier)
-        
+
         XCTAssertTrue(TokenManager.deleteToken(identifier, accessGroup: accessGroup))
-        
+
         let actualToken = keychain?.getObjectForKey(identifier) as? AccessToken
         guard actualToken == nil else {
             XCTAssert(false)
             keychain?.deleteObjectForKey(identifier)
             return
         }
-        
+
         let testCookieStorage = NSHTTPCookieStorage.sharedHTTPCookieStorage()
         XCTAssertEqual(testCookieStorage.cookies?.count, 0)
-        
+
     }
-    
-    
+
+    func testAppDeleteFunctionality() {
+        let identifier = "testIdentifier"
+        let accessGroup = "testAccessGroup"
+
+        let token = getTestToken()
+
+        TokenManager.saveToken(token, tokenIdentifier:identifier, accessGroup: accessGroup)
+
+        let actualToken = TokenManager.fetchToken(identifier, accessGroup: accessGroup)
+        XCTAssertNotNil(actualToken)
+        XCTAssertEqual(actualToken?.tokenString, token.tokenString)
+
+        // When app deleted, cleared the user default
+        NSUserDefaults.standardUserDefaults().removeObjectForKey(kCheckFirstRun)
+        NSUserDefaults.standardUserDefaults().synchronize()
+
+        let deletedToken = TokenManager.fetchToken(identifier, accessGroup: accessGroup)
+        XCTAssertNil(deletedToken)
+        XCTAssertNotEqual(deletedToken?.tokenString, token.tokenString)
+
+        TokenManager.deleteToken(identifier, accessGroup: accessGroup)
+    }
+
+
```

Comment:
Can we also add a test that verifies that the bool is saved to user defaults when a token is saved?